src/main/java/org/folio/coursereserves/util/CRUtil.java

```diff
@@ -299,211 +263,141 @@ public static void populateReserveCopiedItemFromJson(Reserve reserve, JsonObject
-                + "/" + instanceId, HttpMethod.GET, null, null, 200).onComplete(
-                instanceRes -> {
-              if(instanceRes.failed()) {
-                promise.fail(instanceRes.cause());
-              } else {
-                JsonObject instanceJson = instanceRes.result();
-                result.put("instance", instanceJson);
-                logger.info("Inventory lookup complete");
-                promise.complete(result);
-              }
-            });
-          }
+    return makeOkapiRequest(context.owner(), okapiHeaders, ITEMS_ENDPOINT + "/" + itemId,
+        HttpMethod.GET, null, null, 200).compose(itemJson -> {
+      JsonObject result = new JsonObject();
+      String holdingsId = itemJson.getString("holdingsRecordId");
+      result.put("item", itemJson);
+      logger.info("Making request for holdings at {}/{}", HOLDINGS_ENDPOINT, holdingsId);
+      return makeOkapiRequest(context.owner(), okapiHeaders, HOLDINGS_ENDPOINT + "/" + holdingsId,
+          HttpMethod.GET, null, null, 200).compose(holdingsJson -> {
+        String instanceId = holdingsJson.getString("instanceId");
+        result.put("holdings", holdingsJson);
+        logger.info("Making request for instance at {}/{}", INSTANCES_ENDPOINT, instanceId);
+        return makeOkapiRequest(context.owner(), okapiHeaders, INSTANCES_ENDPOINT
+            + "/" + instanceId, HttpMethod.GET, null, null, 200).map(instanceJson -> {
+              result.put("instance", instanceJson);
+              return result;
         });
-      }
+      });
     });
-    return promise.future();
   }
 
   public static Future<JsonObject> lookupUserAndGroupByUserId(String userId,
       Map<String, String> okapiHeaders, Context context) {
-    Promise<JsonObject> promise = Promise.promise();
     String userPath = "/users/" + userId;
-    JsonObject result = new JsonObject();
-    makeOkapiRequest(context.owner(), okapiHeaders, userPath, HttpMethod.GET,
-        null, null, 200).onComplete(userRes -> {
-      try {
-        if(userRes.failed()) {
-          promise.fail(userRes.cause());
-        } else {
-          result.put("user", userRes.result());
-          String groupId = userRes.result().getString("patronGroup");
-          String groupPath = "/groups/" + groupId;
-          makeOkapiRequest(context.owner(), okapiHeaders, groupPath, HttpMethod.GET,
-              null, null, 200).onComplete(groupRes -> {
-            try {
-              if(groupRes.failed()) {
-                promise.fail(groupRes.cause());
-              } else {
-                result.put("group", groupRes.result());
-                promise.complete(result);
-              }
-            } catch(Exception e) {
-              promise.fail(e);
-            }
-          });
-        }
-      } catch(Exception e) {
-        promise.fail(e);
-      }
+    return makeOkapiRequest(context.owner(), okapiHeaders, userPath, HttpMethod.GET,
+        null, null, 200).compose(userRes -> {
```

Comment:
For me it's more readable if there's a line break before each `.compose` and `.map`.